forgit.plugin.zsh

```diff
@@ -63,7 +64,7 @@ forgit::add() {
     "
     files=$(git -c color.status=always -c status.relativePaths=true status --short |
         grep -F -e "$changed" -e "$unmerged" -e "$untracked" |
-        awk '{printf "[%10s]  ", $1; $1=""; print $0}' |
+        sed 's/^\(..[^[:space:]]*\)[[:space:]]\+\(.*\)$/[\1]  \2/' |
```

Comment:
Hi @wfxr, working on this now, but want to be 100% sure I'm understanding what this is before I put it in!

Can you explain what this sed command, and the sed command up top do? I'm not super well versed on sed

Reply:
Here regular expressions are used for capturing and replacing.

If you remove the escape character it will look a bit clearer:
`sed 's/^(..[^[:space:]]*)[[:space:]]+(.*)$/[\1]  \2/'`. 

What this line does is to capture the status indicator at the beginning of the line, surround it with square brackets, then append the full path. Completely explaining sed's regex rules is not easy. But you can test it to see how it works:

``` bash
# go to `forgit`'s local repo
touch 'a          b   c'
echo 'hello world!' >> README.md

# buggy
git status --short | awk '{printf "[%10s]  ", $1; $1=""; print $0}'

# fixed
git status --short | sed 's/^\(..[^[:space:]]*\)[[:space:]]\+\(.*\)$/[\1]  \2/'
```